addons/mail/models/res_partner.py

```diff
@@ -126,14 +126,15 @@ def _message_fetch_failed(self):
         """Returns all messages, sent by the current partner, that have errors, in
         the format expected by the web client."""
         self.ensure_one()
-        messages = self.env['mail.message'].search([
-            ('has_error', '=', True),
+        notifications = self.env['mail.notification'].search([
```

Comment:
Regular people should not have access to mail.notification model. I would sudo to avoid unnecessary acls issues, especially we plan to remove most access to this model.

Reply:
Shouldn't the change be done in the PR changing the access then?
Because if it's currently possible to craft a notification manually, adding sudo here would allow access to messages that shouldn't be accessed.

Reply:
How would that add notifications ? Domain seems sufficiently tailored.

Reply:
With `sudo` and the condition on `message author id` removed (according to the other comment), it becomes possible (if notification access are not changed at the same time) to create a notification targeting any message from anyone else, and receiving (some fields of) this message though this call.
Ok the use case is restricted to message with failures thanks to the other conditions, but that's still a potential breach.

Reply:
We already have some custom check for strange use cases while we continue to support writing on mail.notification object (just remembered it), like
```
    def write(self, vals):
        if ('mail_message_id' in vals or 'res_partner_id' in vals) and not self.env.is_admin():
            raise AccessError(_("Can not update the message or recipient of a notification."))
```
We could decide to do the same for author_id update. If you update a notification's author without being granted (condition to cleanup), it raises.

Idea is that: you can modify some of your notification fields, but not "critical" one that could grant access to information.

Reply:
Oh did not understand your answer. Actually I don't understand how a private method leaks information. This is the role of the caller of _message_something to ensure information is not leaked. Private (low level) methods are generally optimized in sudo as they are not callable.

Reply:
In this case it can't be ensured pre-call as the method is "return all failure of given user" there is nothing to validate/no user input, it's always an "accepted" call. Only the method itself has to determine which failures are correct to return for the given user, and it is currently relying on access domain on both notification and message to do its job.

I really don't want to add a `sudo` that we don't need, generally speaking. And even more so for a reason that depends on a PR that doesn't exist yet (to re-use your argument against the optimization of `related`).

Reply:
Ok, your choice on that.